src/decision/location_selection.jl

```diff
@@ -0,0 +1,266 @@
+
+    rank_frequencies = NamedDimsArray(zeros(mn...); zip(dn_subset, freq_elements)...)
+
+    for rank in 1:n_ranks
+        rank_frequencies[ranks=Int64(rank)] .= sum(ranks .== rank; dims=:scenarios)[scenarios=1]
+    end
+
+    return rank_frequencies
+end
+function ranks_to_frequencies(
+    ranks::NamedDimsArray{D,T,3,A};
+    n_ranks=length(ranks.sites),
+    agg_func=x -> dropdims(sum(x; dims=:timesteps); dims=:timesteps),
+) where {D,T,A}
+    return agg_func(ranks_to_frequencies(ranks, n_ranks))
+end
+function ranks_to_frequencies(
+    ranks::NamedDimsArray{D,T,2,A};
+    n_ranks=length(ranks.sites),
+    agg_func=nothing,
+) where {D,T,A}
+    if !isnothing(agg_func)
+        return agg_func(ranks_to_frequencies(ranks, n_ranks))
+    end
+
+    return ranks_to_frequencies(ranks, n_ranks)
+end
+
+"""
+    location_selection_frequencies(ranks::NamedDimsArray;n_loc_int::Int64=5)
+    location_selection_frequencies(inv_log::NamedDimsArray{D,T,4,A};dims::Union{Symbol,Vector{Symbol}}=:coral_id) where {D,T,A}
+
+Post-processing function for intervention logs. Calculates the frequencies with which locations were selected for a particular intervention,
+for a selection of scenarios (e.g. selected robust scenarios).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `n_loc_int` : number of locations which are intervened at for each intervention decision.
+- `dims` : dimensions to sum selection frequencies over.
+
+# Returns 
+Counts for location selection at each location in the domain.
+"""
+function location_selection_frequencies(
+    ranks::NamedDimsArray;
+    n_loc_int::Int64=5,
+)
+    ranks_frequencies = ranks_to_frequencies(ranks; n_ranks=n_loc_int)
+    loc_count = sum(ranks_frequencies[ranks=1:n_loc_int], dims=2)[ranks=1]
+
+    return loc_count
+end
+function location_selection_frequencies(
+    inv_log::NamedDimsArray{D,T,4,A};
+    dims::Union{Symbol,Vector{Symbol}}=:coral_id,
+) where {D,T,A}
+    loc_count = dropdims(
+        sum(dropdims(sum(inv_log; dims=dims); dims=dims) .> 0; dims=:scenarios);
+        dims=:scenarios,
+    )
+    return loc_count
+end
+
+"""
+    summed_inverse_rank(ranks::NamedDimsArray{D,T,3,A};dims::Union{Symbol,Vector{Symbol}}=[:scenarios, :timsteps],
+            ) where {D,T,A}
+    summed_inverse_rank(ranks::NamedDimsArray{D,T,2,A};) where {D,T,A}
+    summed_inverse_rank(ranks::NamedDimsArray,dims::Union{Symbol,Vector{Symbol}},)
+
+Calculates (number of sites) .- ranks summed over the dimension dims and transformed using agg_func 
+    (default no transformation).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `dims` : Dimensions to sum over.
+
+# Returns 
+Inverse rankings (i.e. the greater the number the higher ranked the site).
+"""
+function summed_inverse_rank(
```

Comment:
@Zapiano are you happy with this function name?

Reply:
To answer your question I need to ask: @Rosejoycrocker what is the purpose of this function? If I understand correctly, a scenario with higher ranks will get a low value and vice-versa, right? So what is this going to be used for?

Reply:
Hey @Zapiano, this is the function you suggested in one of our meetings :)
For `ranks` the lower the rank the `better` the site for intervention. For example, 1 is highest ranked, 2 is second highest etc.
So the inverse ranks function instead gives a higher number for higher ranked sites.
As the values are also summed, it gives a measure of the number of times that site was highly ranked . It is also normalised.
This could be used to plot maps which summarise which sites were ranked highly most frequently with a number from 0-1 and hence can be easily compared.
I intend to submit another PR which includes functions to map these values as heatmaps.

Reply:
Yes, I know is that function, but that was a long time ago, so I'm trying to remember and get context again 😄 

So, this functions name, right now, is kind of a "summary" of the mathematical operations it does (it takes the inverse ranks and then sum it). But what it does "conceptually" sound more like an `agg_selection_score` (or `selection_score`) to me - trying to think if there's a better name for this. Because when I say I have the "aggregated selection score" for all locations, you know that is aggregation information about the selection (for all locations), and the greater the number the better.

What do you think?

Reply:
All good haha, it was a while ago

I like the name `selection_score()`, what do you think @ConnectedSystems ?

Reply:
`selection_score()` makes sense to me